opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/service/PsuAuthService.java

```diff
@@ -0,0 +1,49 @@
+package de.adorsys.opba.tppauthapi.service;
+
+import de.adorsys.datasafe.encrypiton.api.types.UserIDAuth;
+import de.adorsys.opba.db.domain.entity.psu.Psu;
+import de.adorsys.opba.db.repository.jpa.psu.PsuRepository;
+import de.adorsys.opba.protocol.facade.config.encryption.impl.psu.PsuSecureStorage;
+import de.adorsys.opba.tppauthapi.exceptions.PsuAuthenticationException;
+import de.adorsys.opba.tppauthapi.exceptions.PsuRegisterException;
+import de.adorsys.opba.tppauthapi.exceptions.PsuAuthorizationException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PsuAuthService {
+
+    private final PsuRepository psuRepository;
+    private final PsuSecureStorage psuSecureStorage;
+
+    @Transactional(readOnly = true)
+    public Psu tryAuthenticateUser(String psuId, String password) throws PsuAuthorizationException {
+        Optional<Psu> psu = psuRepository.findByUserId(psuId);
+        if (!psu.isPresent()) {
+            throw new PsuAuthenticationException("User not found: " + psuId);
+        }
+        UserIDAuth idAuth = new UserIDAuth(psu.get().getId().toString(), password::toCharArray);
+        try {
+            psuSecureStorage.userProfile().privateProfile(idAuth).getKeystore();
```

Comment:
This does not check the user-keystore password, you want to:
`psuSecureStorage.privateService().read(ReadRequest.forDefaultPrivate(idAuth, "dummy"));`

and when registering user you should write `dummy` into his private space. For now, it can be empty file

Reply:
done